src/test/java/org/takes/facets/auth/social/PsLinkedinTest.java

```diff
@@ -67,100 +60,271 @@
-                    @Override
-                    public Response act(final Request req) throws IOException {
-                        return new RsJson(
-                            Json.createObjectBuilder()
-                                .add("id", identifier)
-                                .add(firstname, frodo)
-                                .add(lastname, baggins)
-                                .build()
-                        );
-                    }
-                }
+                peoplepath,
+                new PeopleTake(identifier, firstname, lastname, frodo, baggins)
             )
         );
         new FtRemote(take).exec(
-            // @checkstyle AnonInnerLengthCheck (100 lines)
-            new FtRemote.Script() {
-                @Override
-                public void exec(final URI home) throws IOException {
-                    final Identity identity = new PsLinkedin(
-                        new Href(
-                            String.format(
-                                "%s/uas/oauth2/accessToken",
-                                home
-                            )
-                        ),
-                        new Href(String.format("%s/v1/people/", home)),
-                        lapp,
-                        lkey
-                    ).enter(
-                        new RqFake("GET", String.format("?code=%s", code))
-                    ).get();
-                    MatcherAssert.assertThat(
-                        identity.urn(),
-                        CoreMatchers.equalTo(
-                            String.format("urn:linkedin:%s", identifier)
-                        )
-                    );
-                    MatcherAssert.assertThat(
-                        identity.properties(),
-                        Matchers.allOf(
-                            Matchers.hasEntry(firstname, frodo),
-                            Matchers.hasEntry(lastname, baggins)
-                        )
-                    );
-                }
-            }
+            new LinkedinScript(
+                code, lapp, lkey, identifier,
+                firstname, lastname, frodo, baggins
+            )
         );
     }
+
+    /**
+     * Take that returns JSON with the authorization token.
+     * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+     * @author Rui Castro (rui.castro@gmail.com)
+     * @version $Id$
+     * @since 1.1
+     */
+    private final class TokenTake implements Take {
+
+        /**
+         * Request path pattern for token endpoint.
+         */
+        private final String tokenpath;
+
+        /**
+         * Linkedin authorization code.
+         */
+        private final String code;
+
+        /**
+         * Linkedin app.
+         */
+        private final String lapp;
+
+        /**
+         * Linkedin key.
+         */
+        private final String lkey;
+
+        /**
+         * Ctor.
+         * @param code Linkedin authorization code.
```

Comment:
@rui-castro there should be no dots, see https://github.com/teamed/qulice/issues/733